spec/request_spec.rb

```diff
@@ -1,31 +1,50 @@
 require 'spec_helper'
 include Apivore::RspecBuilder
 
-describe "Example API", :type => :request do
-
-  apivore_setup '/services/{id}.json', 'get', '200' do
-    {'id' => 1}
-  end
-
-  apivore_setup '/services.json', 'post', '204' do
-    {"_data" => {'name' => 'hello world'}}
-  end
-
-  apivore_setup '/services/{id}.json', 'put', '204' do
-    {'id' => 1}
-  end
-
-  apivore_setup '/services/{id}.json', 'delete', '204' do
-    {'id' => 1}
-  end
-
-  apivore_setup '/services/{id}.json', 'patch', '204' do
-    {'id' => 1}
+describe "Example API", type: :apivore, order: :defined do
+
+  include Apivore::RspecHelpers
+
+  subject { Apivore::SwaggerChecker.instance_for("/swagger-doc.json") }
+
+  context "has valid paths" do
+    it "documents /services.json" do
+      expect(subject).to document(:get, "/services.json", 200)
+    end
+
+    it "documents /services.json" do
+      expect(subject).to document(
+        :post, "/services.json", 204, {'name' => 'hello world'}
+      )
+    end
+
+    it "documents /services/{id}.json" do
+      expect(subject).to document(
+        :get, "/services/{id}.json", 200, {'id' => 1}
+      )
+    end
+
+    it "documents /services/{id}.json" do
+      expect(subject).to document(
+        :put, "/services/{id}.json", 204, {'id' => 1}
+      )
+    end
+
+    it "documents /services/{id}.json" do
+      expect(subject).to document(
+        :delete, "/services/{id}.json", 204, {'id' => 1}
+      )
+    end
+
+    it "documents /services/{id}.json" do
+      expect(subject).to document(
+        :patch, "/services/{id}.json", 204, {'id' => 1}
+      )
+    end
   end
 
-  apivore_setup '/services.json', 'get', '200' do
-    expect(Rails.application).to receive(:call).and_call_original
+  context "and" do
+    it { is_expected.to document_all_paths }
```

Comment:
This generates "should have tested all documented routes" which is an older style. Current best practice is to have active sentences, for example "tests all documented routes". I don't think you can easily change the text `is_expected` generates so the best way to fix this is probably:
```ruby
context 'and' do
  it 'tests all documented routes' do
    expect(subject).to document_all_paths
  end
end
```